examplen.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * JS MergeXML NodeJS usage sample
+ * server-side component
+ *
+ * @package     MergeXML
+ * @author      Vallo Reima
+ * @copyright   (C)2019
+ */
+
+const http = require('http');
+const fs = require('fs');
+const formidable = require('formidable');
+const MergeXML = require('./mergexml.js');
```

Comment:
Whatever node solution you use, it would have to be installable with `npm install mergexml`.

I would something like this: 

```js
const MergeXML = require('mergexml');

// only when using in NodeJS:
const xmldom= require('xmldom');
mergeXML.init({
     xpath: require('xpath'),
     DOMParser: xmldom.DOMParser,
    XMLSerializer = xmldom.XMLSerializer
});

See also below, why this is better to avoid bundling unnecessary modules when used in browsers.

Reply:
The approach is that the wrapper supports an environment and the class adapts to it. So, the user app and class itself should not deal with the modules.

Reply:
The problem is that your module is going to be bundled for browser deployments. The solution I proposed doesn't have that problem (because it moves xmldom and XMLSerializer outside your module.

Reply:
There is a 3rd approach to consider that is probably even better. See package.json [documentation](https://docs.npmjs.com/files/package.json#browser) for the `browser` property. It looks like you can have different entry files for browser and nodejs! 

This means you can create a new entry (`main`) file for NodeJS (e.g. using my suggested code above) and add xmldom and XMLSerializer as proper `dependencies` and take away any difficult configuration from users. Both nodeJS and browser can simply `require(mergexml)` (as your current solution) but without creating huge browser bundles. It is true that the 2 NodeJS modules will be 'installed' (in `node_modules`) in apps that will only use mergexml in browser contexts, but that doesn't matter. What matter is that those modules will not be bundled.

Reply:
I see. Will examine a little the bundler logic.